Shopify Store: featur/hader 만들기

xunxee opened the pull request.

xunxee committed:
Modify: Footer 컴포넌트의 width를 반응형으로 수정
- width가 1920px 이후부터는 늘어나지 않도록 수정

xunxee committed:
Modify: Footer 컴포넌트를 하단의 고정시키기 위해 css를 수정
- App.jsx에 display: flex를 설정
- HomePage 컴포넌트에서는 flex:1을 설정하여 남은 크기 전체를 차지하도록 설정

xunxee committed:
Add: HeaderPage에서 카테고리를 클릭하면 URL이 변경되면서 ListPage 컴포넌트가 렌더링되는 액션을 테스트 및 구현
- HeaderPage 컴포넌트에서는 useNavigate()를 사용하여 handleClickCategories의 인자마다 URL의 경로를 변경하는 액션을 구현
- TitleContainer 컴포넌트에서는 HeaderPage 컴포넌트의 handleClickCategories 함수를 props로 받아 인자를 넘겨주는 액션을 구현

xunxee committed:
Add: ListContainer 컴포넌트의 기본적인 레이아웃을 테스트 및 구현
- ListPage의 기본적인 세션을 flex box를 사용하여 구현

ChoJinmok commented on src/App.jsx:
```diff
@@ -24,8 +26,7 @@ export default function App() {
         <Route path="/" element={<HomePage />} />
         <Route path="/search" element={<ListPage />} />
-        <Route path="/search/new" element={<ListPage />} />
-        <Route path="/search/featured" element={<ListPage />} />
+        <Route path="/search/:category" element={<ListPage />} />
```
이렇게 나누지 않고 하나로 해결하는 방법 찾아보시면 좋을 것 같습니다 ㅎㅎ

ChoJinmok commented on src/Footer/FooterBottomContainer.jsx:
```diff
@@ -7,7 +7,11 @@ const Container = styled.div({
+  width: '90%',
+  maxWidth: '1920px',
   height: '100px',
+  marginRight: 'auto',
+  marginLeft: 'auto',
```
아래에 padding도 그렇고 한줄로 작성하지 않고 나눠서 작성하신 이유가 있을까요?~

xunxee replied:
🫢 제가 놓쳣네요 ㅠㅠ

ChoJinmok commented on src/Footer/FooterPage.jsx:
```diff
@@ -6,8 +6,6 @@ import FooterBottomContainer from './FooterBottomContainer';
-  paddingRight: '1.5rem',
   borderTop: '1px solid #999999',
   color: '#EAEAEA',
   backgroundColor: '#000000',
```
숫자가 중복되는 경우에 `#000`이런식으로 작성 가능해서~ 한번 참고하세용 ㅎㅎ

xunxee replied:
숫자도 축약이 가능하군요? 아예 모르고 있었습니다!
감사합니당

ChoJinmok commented on src/Header/TitleContainer.jsx:
```diff
@@ -54,16 +64,31 @@ export default function TitleContainer() {
         <Item>
-          <Link to="/search">All</Link>
+          <a
+            href="/search"
+            onClick={handleClick('All')}
+          >
+            All
+          </a>
         </Item>
         <Item>
-          <Link to="/search/new">New Arrivals</Link>
+          <a
+            href="/search/new"
+            onClick={handleClick('new')}
+          >
+            New Arrivals
+          </a>
         </Item>
         <Item>
-          <Link to="/search/featured">Featured</Link>
+          <a
+            href="/search/featured"
+            onClick={handleClick('featured')}
+          >
+            Featured
+          </a>
         </Item>
```
이 부분이 반복되서 배열로 만드신 다음에 map으로 포현해주셔도 좋을 거 같아요!

ChoJinmok commented on vercel.json:
```diff
@@ -0,0 +1,8 @@
+      "destination": "/index.html"
+    }
+  ]
+}
```
한 칸 띄워주세용 ㅎㅎ

xunxee replied:
하하하😹
push하고 file change 한번 확인하겠습니당

ChoJinmok commented on src/List/ListPage.jsx:
```diff
@@ -1,7 +1,17 @@
+import ListContainer from './ListContainer';
+
+const Container = styled.div({
+  flex: '1',
```
flex: 1을 해주신 이유가 있나용??

xunxee replied:
없어도 잘 동작하는 부분이군요 ㅠㅠ

ChoJinmok reviewed (commented):
List Page는 새로운 PR에서 작성하시는 게 좋을 거 같습니당!!

xunxee committed:
Refactor: FooterBottomContainer 컴포넌트의 color 정의를 축약형으로 리팩터링
- 같은 숫자나 알파벳으로 반복되는 color 값 정의를 축약하여 작성

xunxee committed:
Refactor: TitleContainer 컴포넌트의 CSS를 리팩터링
- 같은 숫자나 알파벳으로 반복되는 color 값 정의를 축약하여 작성
- Loco의 transition을 :hover에서 밖으로 이동시켜 마우스가 밖으로 이동했을 때도 효과가 일어나도록 설정

xunxee committed:
Refactor: FooterPage 컴포넌트의 color 정의를 축약형으로 리팩터링
- 같은 숫자나 알파벳으로 반복되는 color 값 정의를 축약하여 작성

xunxee committed:
Refactor: FooterTopList 컴포넌트의 Item styled() 함수를 사용하여 반복되는 부분을 리팩터링
- Item 컴포넌트의 props를 사용하여 반복되는 스타일링을 리팩터링 작업

xunxee committed:
Refactor: vercel.json의 에러를 잡기 위해 마지막 한 줄을 추가

xunxee committed:
Refactor: HeaderPage 컴포넌트의 color 정의를 축약형으로 리팩터링
- 같은 숫자나 알파벳으로 반복되는 color 값 정의를 축약하여 작성

xunxee committed:
Refactor: ListPage 컴포넌트의 불필요한 CSS 코드를 리팩터링
- 기존의 불필요한 “flex:1” 코드를 수정
- ListPage 컴포넌트의 backgroundColor를 추가

xunxee committed:
Refactor: TitleContainer 컴포넌트에서 event.target을 event.currentTarget으로 변경
- click 이벤트의 target이 svg태그이므로, 이벤트 핸들러를 바인딩하는 DOM 요소를 나타내는 currentTarget을 사용하여 hadnleClick 함수를 리팩터링
